tests/unit/coverage-middleware-test.js

```diff
@@ -0,0 +1,111 @@
+/*jshint expr: true */
+var rewire = require('rewire');
+var expect = require('chai').expect;
+var middleware = rewire('../../lib/coverage-middleware');
+var _ = require('lodash');
+
+var instances = [];
+
+function JsonReporter(options) {
+  this.type = 'JsonReporter';
+  this.options = options;
+  instances.push(this);
+}
+
+JsonReporter.prototype.report = function(data) {
+  this.reported = data;
+};
+
+function LcovReporter(options) {
+  this.type = 'LcovReporter';
+  this.options = options;
+  instances.push(this);
+}
+
+LcovReporter.prototype.report = function(data) {
+  this.reported = data;
+};
+
+function TeamCityReporter(options) {
+  this.type = 'TeamCityReporter';
+  this.options = options;
+  instances.push(this);
+}
+
+TeamCityReporter.prototype.report = function(data) {
+  this.reported = data;
+};
```

Comment:
I'm sure this can be done more elegantly than copy/pasting these "reporters" here, but it was the quickest solution I could think of without sinon or something like it, which I didn't feel like adding just now.